compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -1280,7 +1280,16 @@ private BType checkListConstructorCompatibility(BType bType, BLangListConstructo
             this.dlog.setNonConsoleDLog();
 
             List<BType> compatibleTypes = new ArrayList<>();
+            boolean erroredExpType = false;
             for (BType memberType : ((BUnionType) bType).getMemberTypes()) {
+                if (memberType == symTable.semanticError) {
```

Comment:
How can there be a semantic error in a union type?

Reply:
One example is when there's a member type that is not defined.

```ballerina
public function main() {
    Foo|int[] x = [1, 2];
}
```

We seem to be logging an `unknown type 'Foo'` error, and using semantic error as the member type.